src/key.rs

```diff
@@ -148,6 +148,26 @@ impl SecretKey {
         }
     }
 
+    /// Converts a `SECRET_KEY_SIZE`- array to a secret key
+    #[inline]
+    pub fn from_inner(data: [u8; 32])-> Result<SecretKey, Error> {
+        match data.len() {
```

Comment:
you're matching on the data len when it's already 32 bytes.

maybe just call `SecretKey::from_slice` instead? (altough I do like using the type system to restrict the size of the slice, especially when you can use `TryFrom` to convert a slice to an array)

Reply:
I may be misunderstanding this, but I think there was an unnecessary `copy_from_slice` in `SecretKey::from_slice` if we actually own a `[u8; 32]`. The current `FromStr` implementation provides an owned `[u8; 32]` from hex which we convert to a secret key, so I added this method.